src/pointarray.cpp

```diff
@@ -61,6 +61,48 @@ struct OctreeNode
             delete children[i];
     }
 
+    bool rayPassesNearOrThrough(
+        const V3d& rayOrigin,
+        const V3d& rayDir) const
```

Comment:
Stylistic quibble: Could you put the function arguments on a single line when they fit?  There's nothing wrong with what you have here of course, other than it being a bit inconsistent with the ambient style of the codebase.